com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs

```diff
@@ -640,7 +640,12 @@ enum ClusterDepthSource : int
         int m_CurrentShadowSortedSunLightIndex = -1;
         HDAdditionalLightData m_CurrentSunLightAdditionalLightData;
         DirectionalLightData m_CurrentSunLightDirectionalLightData;
-        internal Light GetCurrentSunLight() { return m_CurrentSunLight; }
+
+        /// <summary>
+        /// Main directional Light for the HD Render Pipeline.
+        /// </summary>
+        /// <returns>The main directional Light.</returns>
+        public Light GetMainDirectionalLight() => m_CurrentSunLight;
```

Comment:
Nice :) I think it would also make sense to have a similar function in shaders like URP have to get the main light: `GetMainLight()`.

Source: https://blog.unity.com/technology/custom-lighting-in-shader-graph-expanding-your-graphs-in-2019

Reply:
if urp has this function, it would make more sense to put the node in core RP and have it work for both RP
I'll try to do that